website/server/models/group.js

```diff
@@ -741,9 +741,13 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
 
     update.$unset = {[`quest.members.${user._id}`]: 1};
   }
-
   // If user is the last one in group and group is private, delete it
   if (group.memberCount <= 1 && group.privacy === 'private') {
+    // this may be redundant, remove any remaining newMessages for the final user
```

Comment:
I don't think this is necessary. You have it covered in the leave and kick routes.